src/js_results.hpp

```diff
@@ -158,20 +157,21 @@ typename T::Object ResultsClass<T>::create_filtered(ContextType ctx, const U &co
     }
 
     auto query_string = Value::validated_to_string(ctx, args[0], "predicate");
-    auto query = collection.get_query();
+
     auto const &realm = collection.get_realm();
     auto const &object_schema = collection.get_object_schema();
-    DescriptorOrdering ordering;
-    parser::KeyPathMapping mapping;
+    query_parser::KeyPathMapping mapping;
     realm::populate_keypath_mapping(mapping, *realm);
-
-    parser::ParserResult result = parser::parse(query_string);
     NativeAccessor<T> accessor(ctx, realm, object_schema);
-    query_builder::ArgumentConverter<ValueType, NativeAccessor<T>> converter(accessor, &args.value[1], args.count - 1);
-    query_builder::apply_predicate(query, result.predicate, converter, mapping);
-    query_builder::apply_ordering(ordering, query.get_table(), result.ordering, mapping);
-
-    return create_instance(ctx, collection.filter(std::move(query)).apply_ordering(std::move(ordering)));
+    query_parser::ArgumentConverter<ValueType, NativeAccessor<T>> converter(accessor, &args.value[1], args.count - 1);
```

Comment:
If args.count < 2, `ArgumentConverter` will be called with `&Value<T>::from_undefined(ctx)` as 2nd argument.  Is this safe?